library/hpe_icsp_os_deployment.py

```diff
@@ -203,6 +191,7 @@ def main():
             username=dict(required=True, type='str'),
             password=dict(required=True, type='str', no_log=True),
             server_id=dict(required=True, type='str'),
```

Comment:
As you can use either the IP ir the serial to find the server, I think you could leave both as requires=False and check on the code If the user provided one of them.